notebooks/02_Fairing/02_04_fairing_kaniko_cloud_builder.ipynb

```diff
@@ -0,0 +1,263 @@
+    "\n",
+    "namespace = fairing_utils.get_current_k8s_namespace()\n",
+    "\n",
+    "logging.info(f\"Running in aws region {AWS_REGION}, account {AWS_ACCOUNT_ID}\")\n",
+    "logging.info(f\"Running in namespace {namespace}\")\n",
+    "logging.info(f\"Using docker registry {DOCKER_REGISTRY}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Use Kubeflow fairing to build the docker image\n",
+    "\n",
+    "* You will use kubeflow fairing's kaniko builder to build a docker image that includes all your dependencies\n",
+    "  * You use kaniko because you want to be able to run `pip` to install dependencies\n",
+    "  * Kaniko gives you the flexibility to build images from Dockerfiles"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from kubeflow.fairing.builders import cluster\n",
+    "\n",
+    "# output_map is a map of extra files to add to the notebook.\n",
+    "# It is a map from source location to the location inside the context.\n",
+    "output_map =  {\n",
+    "    \"Dockerfile.model\": \"Dockerfile\",\n",
+    "    \"model.py\": \"model.py\"\n",
+    "}\n",
+    "\n",
+    "preprocessor = base_preprocessor.BasePreProcessor(\n",
+    "    command=[\"python\"], # The base class will set this.\n",
+    "    input_files=[],\n",
+    "    path_prefix=\"/app\", # irrelevant since we aren't preprocessing any files\n",
+    "    output_map=output_map)\n",
+    "\n",
+    "preprocessor.preprocess()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a new ECR repository to host model image\n",
+    "!aws ecr create-repository --repository-name fairing-kaniko --region=$AWS_REGION"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Use a Tensorflow image as the base image\n",
+    "# We use a custom Dockerfile \n",
+    "cluster_builder = cluster.cluster.ClusterBuilder(registry=DOCKER_REGISTRY,\n",
+    "                                                 base_image=\"\", # base_image is set in the Dockerfile\n",
+    "                                                 preprocessor=preprocessor,\n",
+    "                                                 image_name=\"fairing-kaniko\",\n",
+    "                                                 dockerfile_path=\"Dockerfile\",\n",
+    "                                                 pod_spec_mutators=[fairing.cloud.aws.add_aws_credentials_if_exists, fairing.cloud.aws.add_ecr_config],\n",
+    "                                                 context_source=cluster.s3_context.S3ContextSource(region=AWS_REGION))\n",
+    "cluster_builder.build()\n",
+    "logging.info(f\"Built image {cluster_builder.image_tag}\")"
```

Comment:
Let's have a another README section to say you can use this image for training, etc. As a user, we only know how to build a image here and I don't know how to use it somewhere. Can you check training job part? If that's no a lot of work, do you think it's better to kick a job here and use the image we just build? What do you think?

Reply:
From my side, I think it's okay to kick off a training job. Let me fix it in the latest PR.